addons/pos_restaurant/models/pos_session.py

```diff
@@ -35,6 +35,11 @@ def _loader_params_restaurant_table(self):
             },
         }
 
+    def _loader_params_account_fiscal_position(self):
+        params = super()._loader_params_account_fiscal_position()
+        params['search_params']['domain'] = [('|'), ('id', 'in', self.config_id.fiscal_position_ids.ids), ('id', '=', self.config_id.take_away_alternative_fp_id.id)]
```

Comment:
```suggestion
        params['search_params']['domain'] = expression.OR([
            params['search_params']['domain'],
            [('id', '=', self.config_id.take_away_alternative_fp_id.id)]
        ])
```

reuse the original domain.